extensions/notebook/src/dialog/configurePython/configurePathPage.ts

```diff
@@ -0,0 +1,191 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import { BasePage } from './basePage';
+import * as nls from 'vscode-nls';
+import { JupyterServerInstallation } from '../../jupyter/jupyterServerInstallation';
+import { PythonPathInfo } from '../pythonPathLookup';
+import * as utils from '../../common/utils';
+
+const localize = nls.loadMessageBundle();
+
+export class ConfigurePathPage extends BasePage {
+	private readonly BrowseButtonText = localize('configurePython.browseButtonText', "Browse");
+	private readonly LocationTextBoxTitle = localize('configurePython.locationTextBoxText', "Python Install Location");
+	private readonly SelectFileLabel = localize('configurePython.selectFileLabel', "Select");
+
+	private pythonLocationDropdown: azdata.DropDownComponent;
+	private pythonDropdownLoader: azdata.LoadingComponent;
+	private browseButton: azdata.ButtonComponent;
+	private newInstallButton: azdata.RadioButtonComponent;
+	private existingInstallButton: azdata.RadioButtonComponent;
+
+	private usingCustomPath: boolean = false;
+
+	public async start(): Promise<boolean> {
+		this.pythonLocationDropdown = this.view.modelBuilder.dropDown()
+			.withProperties<azdata.DropDownProperties>({
+				value: undefined,
+				values: [],
+				width: '100%'
+			}).component();
+		this.pythonDropdownLoader = this.view.modelBuilder.loadingComponent()
+			.withItem(this.pythonLocationDropdown)
+			.withProperties<azdata.LoadingComponentProperties>({
+				loading: false
+			})
+			.component();
+
+		this.browseButton = this.view.modelBuilder.button()
+			.withProperties<azdata.ButtonProperties>({
+				label: this.BrowseButtonText,
+				width: '70px'
+			}).component();
+		this.browseButton.onDidClick(() => this.handleBrowse());
+
+		this.createInstallRadioButtons(this.view.modelBuilder, this.model.useExistingPython);
+
+		let formModel = this.view.modelBuilder.formContainer()
+			.withFormItems([{
+				component: this.newInstallButton,
+				title: localize('configurePython.installationType', "Installation Type")
+			}, {
+				component: this.existingInstallButton,
+				title: ''
+			}, {
+				component: this.pythonDropdownLoader,
+				title: this.LocationTextBoxTitle
+			}, {
+				component: this.browseButton,
+				title: ''
+			}]).component();
+
+		await this.view.initializeModel(formModel);
+
+		await this.updatePythonPathsDropdown(this.model.useExistingPython);
+
+		return true;
+	}
+
+	public async onPageEnter(): Promise<boolean> {
+		return true;
```

Comment:
What do you expect to happen if this returns false? I would expect the logic for determining if you could navigate to the next page should all be in the onPageLeave methods